mrtt-ui/src/components/SiteInterventions/MangroveAssociatedSpeciesRow.js

```diff
@@ -0,0 +1,130 @@
+import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { Box, MenuItem, TextField } from '@mui/material'
+import { Delete } from '@mui/icons-material'
+import { styled } from '@mui/material/styles'
+
+import { TabularInputSection, TabularLabel, VerticalTabularBox } from '../../styles/forms'
+import { purposeOptions, sourceOptions } from '../../data/siteInterventionOptions'
+
+const MangroveAssociatedSpeciesRow = ({
+  type,
+  count,
+  source,
+  purpose,
+  other,
+  index,
+  deleteItem,
+  updateItem
+}) => {
+  const [initialCount, setInitialCount] = useState('')
+  const [currentCount, setCurrentCount] = useState('')
+  const [initialSource, setInitialSource] = useState('')
+  const [currentSource, setCurrentSource] = useState('')
+  const [initialPurpose, setInitialPurpose] = useState('')
+  const [currentPurpose, setCurrentPurpose] = useState('')
+  const [initialOther, setInitialOther] = useState('')
+  const [currentOther, setCurrentOther] = useState('')
+
+  const handleDelete = () => {
+    deleteItem(index)
+  }
+
+  const handleUpdate = () => {
+    if (
+      currentCount !== initialCount ||
+      currentSource !== initialSource ||
+      currentPurpose !== initialPurpose ||
+      currentOther !== initialOther
```

Comment:
I might be a bit confused here, but won't values on both side of the comparator always be the same based on the useEffect below this?